src/lexer.cpp

```diff
@@ -0,0 +1,109 @@
+#include <algorithm>
+#include <iostream>
+#include <iterator>
+#include <cstdlib>
+#include <string>
+#include <vector>
+
+#include "lexer.h"
+
+//VARIABLES
+std::vector<std::string> all_types = {"PROGRAM", "VAR", "END_VAR", "END_PROGRAM", "BOOL", "TRUE", "FALSE"}; //keywords
+std::vector<std::string> operators = {"<", ">", "+", "-", "%", ":="}; // operators
+std::vector<std::string> all_sc = {"{", "}", "[", "]", ";", ":"}; // special characters
+std::vector<Token> fin_tokens;
+
+void display(){
+    for(int i = 0; i < fin_tokens.size(); i++){
+        fin_tokens[i].print();
+    }
+}
+
+void tokenizerFunc (std::string fileInput){
```

Comment:
Rather than having a global variable, this function should return the token vector. Also, it should have a verb name such as "tokenize"
```suggestion
void tokenize (std::string fileInput){
```